src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.ts

```diff
@@ -410,16 +431,38 @@ export class ProgrammingExerciseUpdateComponent implements OnInit {
             }
         });
 
-        this.supportsJava = this.programmingLanguageFeatureService.supportsProgrammingLanguage(ProgrammingLanguage.JAVA);
-        this.supportsPython = this.programmingLanguageFeatureService.supportsProgrammingLanguage(ProgrammingLanguage.PYTHON);
-        this.supportsC = this.programmingLanguageFeatureService.supportsProgrammingLanguage(ProgrammingLanguage.C);
-        this.supportsHaskell = this.programmingLanguageFeatureService.supportsProgrammingLanguage(ProgrammingLanguage.HASKELL);
-        this.supportsKotlin = this.programmingLanguageFeatureService.supportsProgrammingLanguage(ProgrammingLanguage.KOTLIN);
-        this.supportsVHDL = this.programmingLanguageFeatureService.supportsProgrammingLanguage(ProgrammingLanguage.VHDL);
-        this.supportsAssembler = this.programmingLanguageFeatureService.supportsProgrammingLanguage(ProgrammingLanguage.ASSEMBLER);
-        this.supportsSwift = this.programmingLanguageFeatureService.supportsProgrammingLanguage(ProgrammingLanguage.SWIFT);
-        this.supportsOCaml = this.programmingLanguageFeatureService.supportsProgrammingLanguage(ProgrammingLanguage.OCAML);
-        this.supportsEmpty = this.programmingLanguageFeatureService.supportsProgrammingLanguage(ProgrammingLanguage.EMPTY);
+        this.supportedLanguages = [];
+
+        if (this.programmingLanguageFeatureService.supportsProgrammingLanguage(ProgrammingLanguage.JAVA)) {
```

Comment:
Possible follow up refactor: You could use a string literal type enum to have all programming languages in an array of typed strings.
see https://stackoverflow.com/questions/45626602/coverting-typescript-string-literal-types-to-an-array-of-string
```ts
const programmingLanguages = ['Java', 'Python', 'C', ... ,'Haskell'] as const
type ProgrammingLanguages = typeof programmingLanguages[number]
```

Using that you could do
```ts
programmingLanguages
  .filter(lang => this.programmingLanguageFeatureService.supportsProgrammingLanguage(lang))
  .forEach(lang => this.supportedLanguages.push(lang)
```

Something similar might be possible with enums but using enums is discouraged on most cases